node_settled_handler.go

```diff
@@ -115,51 +110,61 @@ func (p *NodeSettledHandler) subscribeEvents(ctx context.Context) error {
 		if !ok {
 			continue
 		}
-		if !finalEvent.FinalHtlcEvent.Settled {
-			// TODO: Handle final fail?
-
-			continue
-		}
 
 		key := types.CircuitKey{
 			ChanID: event.IncomingChannelId,
 			HtlcID: event.IncomingHtlcId,
 		}
 
-		if err := p.handleFinalHtlc(ctx, key); err != nil {
+		err = p.handleFinalHtlc(
+			ctx, key, finalEvent.FinalHtlcEvent.Settled,
+		)
+		if err != nil {
 			return err
 		}
 	}
 }
 
 func (p *NodeSettledHandler) handleFinalHtlc(ctx context.Context,
-	key types.CircuitKey) error {
+	key types.CircuitKey, settled bool) error {
+
+	var status types.Status
+	if settled {
+		status = types.StatusSettled
+	} else {
+		status = types.StatusFailed
+	}
 
 	htlcKey := types.HtlcKey{
 		ChanID: key.ChanID,
 		HtlcID: key.HtlcID,
 		Node:   p.lnd.PubKey(),
 	}
 
-	settledHash, err := p.persister.MarkHtlcSettled(ctx, htlcKey)
+	invoiceHash, err := p.persister.MarkHtlcFinal(ctx, htlcKey, settled)
 	switch {
-	case err == persistence.ErrHtlcAlreadySettled:
+	case err == persistence.ErrHtlcAlreadyFinal:
 		return nil
 
-	// If the htlc is not found, the final resolution was for an htlc that isn't
-	// managed by lnmux.
+	// If the htlc is not found, the final resolution was for an htlc that
+	// isn't managed by lnmux.
 	case err == types.ErrHtlcNotFound:
 		return nil
 
 	case err != nil:
-		return fmt.Errorf("unable to mark htlc %v settled: %w", key, err)
+		return fmt.Errorf("unable to mark htlc %v as %s : %w",
+			key, status, err)
 	}
 
-	p.logger.Infow("Htlc final settled received",
-		"chanID", key.ChanID, "htlcID", key.HtlcID, "hash", settledHash)
+	p.logger.Infow("Htlc final resolution received",
+		"chanID", key.ChanID,
+		"htlcID", key.HtlcID,
+		"hash", invoiceHash,
+		"status", status,
+	)
 
-	if settledHash != nil {
-		p.settledCallback(*settledHash, types.StatusSettled)
+	if invoiceHash != nil {
+		p.settledCallback(*invoiceHash, status)
```

Comment:
After a second thought, maybe it makes more sense here indeed.
More strict, and it makes sense that `settleHandler.InvoiceSettled` sends a bool in the channel instead of a status.

Then the server, depending on the bool, will return the right status (I really prefer to send the status (and not the bool) through the RPC call, it will be confusing the the user otherwise)